engine/dlib/src/test/test_opaque_handle_container.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020-2023 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+// 
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+// 
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "dlib/opaque_handle_container.h"
+
+#define JC_TEST_IMPLEMENTATION
+#include <jc_test/jc_test.h>
+
+TEST(dmOpaqueHandleContainer, Basic)
+{
+	dmOpaqueHandleContainer<int> simple;
+	ASSERT_EQ(simple.Capacity(), 0);
+	ASSERT_TRUE(simple.Full());
+
+	// Container has no storage
+	ASSERT_EQ(simple.Put(0), INVALID_OPAQUE_HANDLE);
```

Comment:
Do we want a different behavior from array+hashtable, which assert on this operation?